File: New folder/python/simpleTree.py
class node:
    def __init__(self, data, left=None, right=None):
        self.data = data
        self.left = left
        self.right = right

def preorder(root):
    '''Performs the preorder traversal of the Binary Tree'''
    if root:
        print(root.data,end=' ')
        preorder(root.left)
        preorder(root.right)

def inorder(root):
    '''Performs the inorder traversal of the Binary Tree'''
    if root:
        inorder(root.left)
        print(root.data,end=' ')
        inorder(root.right)

def postorder(root):
    '''Performs the postorder traversal of the Binary Tree'''
    if root:
        postorder(root.left)
        postorder(root.right)
        print(root.data,end=' ')

def main():
    root = node(5)
    root.left = node(4)
    root.right = node(3)
    root.right.left = node(2)
    root.left.right = node(1)
    root.right.right = node(7)
    root.left.left = node(6)

    print("Preorder: ",end = '')
    preorder(root)
    print()
    print("Inorder: ",end = '')
    inorder(root)
    print()
    print("Postorder: ",end = '')
    postorder(root)
    print()


if __name__=="__main__":
    main()

File: New folder/cpp/simpleTree.cpp
#include <bits/stdc++.h>
#define ll long long
#define ld long double
#define forLoop(n) for(int i=0;i<n;i++)
#define space ' '
#define newline endl

using namespace std;

class node{
    public:
        int data;
        node *left,*right;
        node(int data,node*left=NULL,node*right=NULL):data{data},left{left},right{right}{}
};
 
void preOrder(node *root){
    if(!root) return;
    cout << root->data << space;
    preOrder(root->left);
    preOrder(root->right);
}

void inOrder(node *root){
    if(!root) return;
    inOrder(root->left);
    cout << root->data << space;
    inOrder(root->right);
}

void postOrder(node *root){
    if(!root) return;
    postOrder(root->left);
    postOrder(root->right);
    cout << root->data << space;
}

int main()
{
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);

    node *root=new node(5);
    root->left=new node(4);
    root->right=new node(3);
    root->left->left=new node(2);
    root->left->right=new node(1);
    root->right->left=new node(6);
    root->right->right=new node(0);

    cout << "Preorder: ";
    preOrder(root);
    cout << newline << "Inorder: ";
    inOrder(root);
    cout << newline << "Postorder: ";
    postOrder(root);
    cout << newline;

    return 0;
}